lan867x/examples/lan867x_tcp_sockets/server/main/server.c

```diff
@@ -69,26 +76,52 @@ void app_main(void)
     // Stop dhcp client and set a static IP address
     esp_netif_dhcpc_stop(eth_netif);
     esp_netif_set_ip_info(eth_netif, &ip_info);
+#if CONFIG_EXAMPLE_LAN867X_USE_PLCA
+    // Configure PLCA as coordinator
+    uint8_t plca_nodes_count = CONFIG_EXAMPLE_LAN867X_PLCA_NODE_COUNT;
+    esp_eth_ioctl(eth_handles[0], LAN867X_ETH_CMD_S_PLCA_NCNT, &plca_nodes_count);
+    uint8_t plca_id = 0;
+    esp_eth_ioctl(eth_handles[0], LAN867X_ETH_CMD_S_PLCA_ID, &plca_id);
+    uint8_t plca_max_burst_count = 0;
+    esp_eth_ioctl(eth_handles[0], LAN867X_ETH_CMD_S_MAX_BURST_COUNT, &plca_max_burst_count);
+    bool plca_en = true;
+    esp_eth_ioctl(eth_handles[0], LAN867X_ETH_CMD_S_EN_PLCA, &plca_en);
+#endif // otherwise rely on CSMA/CD
     // Start Ethernet driver state machine
     ESP_ERROR_CHECK(esp_eth_start(eth_handles[0]));
     // Initialize Berkley socket which will listen on port SOCKET_PORT for transmission from client
     char rxbuffer[SOCKET_MAX_LENGTH] = {0};
     int server_fd = socket(AF_INET, SOCK_STREAM, 0);
     struct sockaddr_in address;
     socklen_t addrlen = sizeof(address);
+    fd_set ready;
+    struct timeval to = {.tv_sec = 0, .tv_usec = 500000};
     int opt = 1;
     setsockopt(server_fd, SOL_SOCKET, SO_REUSEADDR | SO_REUSEPORT, &opt, sizeof(opt));
     address.sin_family = AF_INET;
     address.sin_addr.s_addr = INADDR_ANY;
     address.sin_port = htons(SOCKET_PORT);
     bind(server_fd, (struct sockaddr *)&address, sizeof(address));
     // listen and wait for transmission to come
+    struct connection_info connections[LISTENER_MAX_QUEUE];
+    int active_connections_count = 0;
     listen(server_fd, LISTENER_MAX_QUEUE);
-    int new_socket = accept(server_fd, (struct sockaddr *)&address, &addrlen);
     while (1) {
-        // when transmission comes - print it
-        recv(new_socket, rxbuffer, SOCKET_MAX_LENGTH, 0);
-        printf("Received: \"%s\" from %s.\n", rxbuffer, inet_ntoa(address.sin_addr));
+        // Check if any connections are pending
+        FD_ZERO(&ready);
+        FD_SET(server_fd, &ready);
+        select(server_fd + 1, &ready, 0, 0, &to);
+        if (FD_ISSET(server_fd, &ready)) {
+            // accept new connection
+            struct sockaddr_in *current_address_ptr = &connections[active_connections_count].address;
+            connections[active_connections_count].fd = accept(server_fd, (struct sockaddr *) current_address_ptr, &addrlen);
+            active_connections_count++;
```

Comment:
You should limit number of active connections since you `connections` has maximum size of `LISTENER_MAX_QUEUE`.